src/replica_ctrl.py

```diff
@@ -155,31 +128,74 @@ def create_mongo_config(tasks_ips, replicaset_name, mongo_port):
     return config
 
 
-def update_config(primary_ip, current_ips, new_ips, replicaset_name, mongo_port):
+def gather_configured_members_ips(mongo_tasks_ips, mongo_port):
+    current_ips = set()
+    for t in mongo_tasks_ips:
+        mc = pm.MongoClient(t, mongo_port)
+        try:
+            config = mc.admin.command("replSetGetConfig")['config']
+            for m in config['members']:
+                current_ips.add(m['host'].split(":")[0])
+        finally:
+            mc.close()
     logger = logging.getLogger(__name__)
-    if not new_ips.difference(current_ips) and not current_ips.difference(new_ips):
-        return
-    else:
-        # Actually not too different from what mongo does:
-        # https://github.com/mongodb/mongo/blob/master/src/mongo/shell/utils.js
-        to_remove = set(current_ips) - set(new_ips)
-        to_add = set(new_ips) - set(current_ips)
-        assert to_remove or to_add
-
-        logger.info("To remove {}".format(to_remove))
-        logger.info("To add {}".format(to_add))
-
-        force = False
-        if (primary_ip in to_remove) or primary_ip is None:
-            logger.info("Primary {} not available".format(primary_ip))
-            force = True
-            #let's see if a new primary was elected
-            primary_ip = get_primary(list(new_ips), replicaset_name, mongo_port)
-            #if not let's find the first mongo that is member of the old cluster
-            if primary_ip is None:
-                primary_ip = list(set(new_ips)-set(to_add))[0]
-
-        cli = pm.MongoClient(primary_ip, mongo_port)
+    logger.info("Current members in mongo configurations: {}".format(current_ips))
+    return current_ips
+
+
+def get_primary_ip(tasks_ips, mongo_port):
+    logger = logging.getLogger(__name__)
+    logger.info("Searching primary")
+
+    primary_ips = []
+    for t in tasks_ips:
+        mc = pm.MongoClient(t, mongo_port)
+        try:
+            if mc.is_primary:
+                primary_ips.append(t)
+        finally:
+            mc.close()
+
+    if len(primary_ips) > 1:
+        logger.error("ERROR: Multiple primaries were found. I.e, replicaset was split")
+        sys.exit(1)
```

Comment:
Fixed. I'll leave the warning for now until I test it better. I'll organize all logging output in a specific branch